src/varint_tests.rs

```diff
@@ -166,4 +166,13 @@ mod tests {
         let mut read = &buf[..];
         assert!(read.read_varint::<u64>().is_err());
     }
+
+    #[test]
+    fn test_varint_with_extra_bytes() {
+        let mut encoded = 0x12345usize.encode_var_vec();
+        assert_eq!(usize::decode_var(&encoded[..]), Some((0x12345, 3)));
+
+        encoded.push(0x99);
+        assert_eq!(usize::decode_var(&encoded[..]), Some((0x12345, 3)));
```

Comment:
would you mind adding another short case testing `usize::decode_var(...) == None`? I.e., if there is no byte with MSB not set. I believe the test just before this one tests it too, but with a Reader.

Thank you! LGTM otherwise.

Reply:
Good idea, done!